src/storage/backend/sqlite/SQLite.cc

```diff
@@ -2,13 +2,58 @@
 
 #include "zeek/storage/backend/sqlite/SQLite.h"
 
+#include <thread>
+
 #include "zeek/3rdparty/sqlite3.h"
 #include "zeek/Func.h"
 #include "zeek/Val.h"
 #include "zeek/storage/ReturnCode.h"
 
+using namespace std::chrono_literals;
+
 namespace zeek::storage::backend::sqlite {
 
+OperationResult SQLite::RunPragma(std::string_view name, std::optional<std::string_view> value) {
+    int attempts = 0;
+    char* errorMsg = nullptr;
+    std::chrono::milliseconds time_spent = 0ms;
+    std::chrono::milliseconds backoff = 1ms;
+
+    std::string cmd = util::fmt("pragma %.*s", static_cast<int>(name.size()), name.data());
+    if ( value && ! value->empty() )
+        cmd += util::fmt(" = %.*s", static_cast<int>(value->size()), value->data());
+
+    while ( pragma_timeout > 0ms && time_spent < pragma_timeout ) {
```

Comment:
```suggestion
    while ( pragma_timeout == 0ms || time_spent < pragma_timeout ) {
```

If pragma_timeout is actually `0` this won't run at all which isn't what the comment in the script layer says :-/ Mind adding a smoke test for this case?